Fix grouping of tables in navigation tree if `$cfg['NavigationTreeTableLevel'] > 1`

lrb2 opened the pull request:
### Description

The contents in the `if` statement were not being run for nodes that have a real parent (ex. a database) but do not have siblings in the navigation tree. This caused an issue where, if `$cfg['NavigationTreeTableLevel']` is set to a value greater than the default of 1 (so that nested groups can exist), group nodes in the navigation tree that do not have a sibling would not be shown. See the pictures below: note that the group `with` was incorrectly not being shown.

Before fix:
![image](https://user-images.githubusercontent.com/59065888/129142795-d8829951-1c53-4459-99f9-f46f2409a1eb.png)

After fix:
![image](https://user-images.githubusercontent.com/59065888/129312743-63afc527-ddaa-4691-beab-ed5fead7ef27.png)



Fixes: #17003

williamdes commented on libraries/classes/Navigation/NavigationTree.php:
```diff
@@ -1143,9 +1143,14 @@ private function renderNode(Node $node, string $class = ''): string
         $paths = $node->getPaths();
         $nodeIsContainer = $node->type === Node::CONTAINER;
-        $hasSiblingsOrIsNotRoot = $node->hasSiblings() || $node->realParent() === false;
+        $hasSiblingsOrIsNotRoot = $node->hasSiblings() || count($node->parents(false, true));
```
could you add a comment and make the count check a bit more explicit with a `> 0` ?

williamdes commented on libraries/classes/Navigation/NavigationTree.php:
```diff
@@ -1143,9 +1143,14 @@ private function renderNode(Node $node, string $class = ''): string
 
+        // Don't show the 'Tables' node under each database unless necessary
+        if ($node instanceof NodeTableContainer && ! $node->hasSiblings()) {
+            $hasSiblingsOrIsNotRoot = false;
```
```suggestion
            // It was not siblings and is a table container so it is self contained
            $hasSiblingsOrIsNotRoot = false;
```
Normally I am not too much of a commenting code person but in this class it is more than necessary for people that will fix issues with it in 5-10 years

I commented about how I understand this if ;)

lrb2 committed:
Fix grouping of tables in navigation tree:
Fix conditions for showing a node in the tree, which caused an issue
where the group name of any group of tables that does not have a sibling
would not be shown in the navigation tree.

Signed-off-by: Lukas Bixen <lrb2_@outlook.com>

lrb2 commented:
The purpose of `$hasSiblingsOrIsNotRoot` was really confusing to me until I went through a bunch of trial-and-error, so I went ahead and renamed the variable to `$showNode`, since that better states what it does. Also, this fixes that issue in  does **not** fix #17003 - this only fixes the issue I casually mentioned at the top of https://github.com/phpmyadmin/phpmyadmin/issues/17003#issuecomment-897056868, though I do have a working solution for #17003 that I'm not completely satisfied with yet.

williamdes commented:
Did you intend to target QA in each of your PRs ?

williamdes reviewed (approved):
It's so much more clear with the new comments and variable rename, thank you !

lrb2 commented:
> Did you intend to target QA in each of your PRs ?

This commit needs to be merged into `master` because the `renderNode` function was just refactored in 41c478edd722c19698db71872ec056eb40f18483, so there would have to be a separate fix only for the QA branch. Though I did switch #17068 to QA, and it looks like I can target QA for #17078, which helps mitigate this issue on its own.

mauriciofauth commented:
Merged, thanks for your contribution!
